ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -756,23 +756,58 @@ private void verifyReadFromFile(NavigableMap<MockId, NavigableSet<IndexValue>> r
     // read from file (unmapped) and verify that everything is ok
     Journal journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     IndexSegment fromDisk = createIndexSegmentFromFile(file, false, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, expectedSizeWritten, false, endOffset,
-        lastModifiedTimeInMs, resetKey);
-    verifyFind(referenceIndex, fromDisk);
-    verifyGetEntriesSince(referenceIndex, fromDisk);
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, false,
+        endOffset, lastModifiedTimeInMs, resetKey);
     // journal should contain all the entries
     verifyJournal(referenceIndex, journal);
     fromDisk.map(true);
 
     // read from file (mapped) and verify that everything is ok
     journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     fromDisk = createIndexSegmentFromFile(file, true, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, expectedSizeWritten, true, endOffset,
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, true,
+        endOffset, lastModifiedTimeInMs, resetKey);
+    // journal should not contain any entries
+    assertNull("Journal should not have any entries", journal.getFirstOffset());
+
+    // delete the bloom file
+    File bloomFile = new File(file.getParent(),
+        IndexSegment.generateIndexSegmentFilenamePrefix(startOffset) + IndexSegment.BLOOM_FILE_NAME_SUFFIX);
+    assertTrue("File could not be deleted", bloomFile.delete());
+    assertFalse("Bloom file should not exist", bloomFile.exists());
+
+    // read from file (mapped) again and verify that everything is ok
+    journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
+    fromDisk = createIndexSegmentFromFile(file, true, journal);
+    assertTrue("Bloom file does not exist", bloomFile.exists());
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, true,
```

Comment:
Does this test if the `bloomFilter.contains()` returns true for all the entries in the index segment? If not, could that be done?

Reply:
It indirectly tests it via the find call (the find call would return `false` if the bloom said it doesn't exist and we would see a failure)

Reply:
Ah, okay, if we are verifying `find()` for every key that verifies the bloom filter as well.